utils/cloudinfo/cloudinfo.go

```diff
@@ -23,19 +23,23 @@ import (
 type CloudInfo interface {
 	GetCloudProvider() info.CloudProvider
 	GetInstanceType() info.InstanceType
+	GetInstanceID() info.InstanceID
 }
 
 type realCloudInfo struct {
 	cloudProvider info.CloudProvider
 	instanceType  info.InstanceType
+	instanceID    info.InstanceID
 }
 
 func NewRealCloudInfo() CloudInfo {
 	cloudProvider := detectCloudProvider()
 	instanceType := detectInstanceType(cloudProvider)
+	instanceID := detectInstanceID(cloudProvider)
```

Comment:
Can this take a long time? (e.g. is it polling ip addresses that may not be always available?)
Would that be a problem? Should this be async?